### ISSUE TITLE
Self hosted / Docker: MSSQL not initialized properly - "vault" not created

### ISSUE THREAD

**Issue by jvalskis** [bitwarden/server#971]
Hello,

MSSQL does not seem to be initializing properly - "vault" database is not created/started(?). See log excerpt from mssql container down below.
Container finishes initializing and other containers try connecting to the DB to which they get the following response: ```Login failed for user 'sa'. Reason: Failed to open the explicitly specified database 'vault'. [CLIENT: 192.168.32.8]```.

Eventually I managed to get the DB up by running ```./bitwarden.sh updatedb```. But this was not a straightforward conclusion to arrive to.

Seeing how noone else is complaining about this, points to the issue being somewhere on my end. But the whole thing is containerized... So I'm at a loss here.

```
2020-10-20 23:20:07.91 spid6s      Starting up database 'msdb'.
2020-10-20 23:20:07.91 spid9s      Starting up database 'mssqlsystemresource'.
2020-10-20 23:20:07.98 spid9s      The resource database build version is 14.00.3257. This is an informational message only. No user action is required.
2020-10-20 23:20:08.38 spid9s      Starting up database 'model'.
2020-10-20 23:20:08.69 spid19s     A self-generated certificate was successfully loaded for encryption.
2020-10-20 23:20:08.71 spid19s     Server is listening on [ 'any' <ipv6> 1433].
2020-10-20 23:20:08.71 spid19s     Server is listening on [ 'any' <ipv4> 1433].
2020-10-20 23:20:08.72 spid19s     Dedicated administrator connection support was not started because it is disabled on this edition of SQL Server. If you want to use a dedicated administrator connection, restart SQL Server using the trace flag 7806. This is an informational message only. No user action is required.
2020-10-20 23:20:08.73 spid19s     SQL Server is now ready for client connections. This is an informational message; no user action is required.
2020-10-20 23:20:09.68 spid9s      Polybase feature disabled.
2020-10-20 23:20:09.68 spid9s      Clearing tempdb database.
2020-10-20 23:20:10.75 spid9s      Starting up database 'tempdb'.
2020-10-20 23:20:11.12 spid20s     The Service Broker endpoint is in disabled or stopped state.
2020-10-20 23:20:11.12 spid20s     The Database Mirroring endpoint is in disabled or stopped state.
2020-10-20 23:20:11.20 spid20s     Service Broker manager has started.
2020-10-20 23:20:11.63 spid6s      Recovery is complete. This is an informational message only. No user action is required.
2020-10-20 23:22:04.03 Logon       Error: 18456, Severity: 14, State: 38.
2020-10-20 23:22:04.03 Logon       Login failed for user 'sa'. Reason: Failed to open the explicitly specified database 'vault'. [CLIENT: 192.168.32.8]
2020-10-21 00:00:01.10 Backup      Error: 3041, Severity: 16, State: 1.
2020-10-21 00:00:01.10 Backup      BACKUP failed to complete the command BACKUP DATABASE vault. Check the backup application log for detailed messages.
2020-10-21 06:20:22.75 Logon       Error: 18456, Severity: 14, State: 38.
2020-10-21 06:20:22.75 Logon       Login failed for user 'sa'. Reason: Failed to open the explicitly specified database 'vault'. [CLIENT: 192.168.32.8]
2020-10-21 06:20:28.10 Logon       Error: 18456, Severity: 14, State: 38.
2020-10-21 06:20:28.10 Logon       Login failed for user 'sa'. Reason: Failed to open the explicitly specified database 'vault'. [CLIENT: 192.168.32.8]
```

After running ```./bitwarden.sh updatedb```:
```
2020-10-21 22:44:03.23 spid51      Starting up database 'vault'.
2020-10-21 22:44:03.44 spid51      Parallel redo is started for database 'vault' with worker pool size [2].
2020-10-21 22:44:03.54 spid51      Parallel redo is shutdown for database 'vault' with worker pool size [2].
2020-10-21 22:44:04.05 spid51      Setting database option AUTO_CLOSE to OFF for database 'vault'.
```

**Comment by Greenderella** [comment]
Hi @jvalskis,

That was all just part of the update. It can be a little confusing the first couple of times you do it. Are you having any issues otherwise after running both update commands (https://bitwarden.com/help/article/updating-on-premise/)?

**Comment by jvalskis** [comment]
Hi @Greenderella, 

I followed the guide (https://bitwarden.com/help/article/install-on-premise/) and it did not explicitly state that update is a necessary action after starting (and it probably isn't, see below). I also ran your commands, but it did not help. 

Howeverm I did manage to get everything up and running after I allowed "public" network to access the internet - seems like that was the source of all of my issues.

I'll try to narrow down the containers that actually need to access the internet and add explicit rules for them (seems like admin and icons most definitely do), but that is out of scope of this issue.

Thanks. Closing the issue.
